coralproject/talk
# [CORL-2384] Upgrade to Node 18

## What does this PR do?

Upgrades Coral to use Node 18 with npm 8.0.0.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- install Node `v18.16.0` with npm `v8.0.0`
  - recommend using `nvm` for this
- run `sh scripts/clear-node_modules.sh`
  - this removes all existing node 14 module files that can cause conflicts
- run `sh npm-i.sh`
  - alternately, run `npm initialize.sh` if you've never run a build before or want to start fresh
- open a terminal in `client/` and run `npm run watch`
- open a terminal in `server/ and run `npm run watch`
- wait for the builds to finish
- navigate to http://localhost:8080 and test out the stream/admin
  - alternatively, do a server only build with `multi-site-test` if that's more to your taste
  
Extra testing:
- create an image using the `Dockerfile`
- spin up the image and see that it runs properly

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.
